dbos/_sys_db.py

```diff
@@ -576,6 +581,11 @@ def resume_workflow(self, workflow_id: str) -> None:
                     status=WorkflowStatusString.ENQUEUED.value,
                     recovery_attempts=0,
                     workflow_deadline_epoch_ms=None,
+                    application_version=(
+                        application_version
+                        if application_version is not None
+                        else original_application_version
```

Comment:
Good fix. But I don't think resume should change the app version. It's confusing to have workflow steps running on different versions.